src/main/scala/machines/Internal.scala

```diff
@@ -3,4 +3,31 @@ package machines
 import regex._
 import dfa._
 
-// TODO: Add your code below
+given Conversion[Char, RegularLanguage] =
+    c => Character(c)
+
+given Conversion[String, RegularLanguage] = 
+    s => s.map(c => Character(c)).reduce(Concat(_,_))
+
+given Conversion[RegularLanguage, DFA] =
+    l => regexToDFA(l, findCharacters(l))
```

Comment:
Interesting approach, I used an apply function calling the toDFA extension which I think was more challenging to wrap my head around than this explicit call to regexToDFA. I think having seen this syntax for conversions would have been very useful